integration/shareddata/shareddata.go

```diff
@@ -85,6 +86,31 @@ func AllProviders() []Provider {
 	return maps.Values(res)
 }
 
+// Providers are array of providers.
+type Providers []Provider
+
+// Remove specified providers and return remaining providers.
+func (ps Providers) Remove(removeProviderNames ...string) Providers {
+	res := make([]Provider, 0, len(ps))
+
+	for _, p := range ps {
+		keep := true
+
+		for _, name := range removeProviderNames {
+			if p.Name() == name {
+				keep = false
+				break
```

Comment:
```suggestion
continue
```

Reply:
Maybe I'm misunderstand, if I use `continue` here this will `continue` in the inner loop. I would like to stop iterating on inner loop. :thinking:

Reply:
ah! yes, you are right! sorry, I misread it